distribution/lib/Standard/AWS/0.0.0-dev/src/S3/S3_File.enso

```diff
@@ -155,13 +203,18 @@ type S3_File
         trimmed = if subpath.starts_with "/" then subpath.drop (First 1) else subpath
         parts = trimmed.split "/"
 
+        extend current part =
+            if current == "" then part else
+                if current.ends_with "/" then current + part else
+                    current + "/" + part
+
         loop current remaining = if remaining.length == 0 then current else
             new_current = case remaining.first of
                 ".." ->
-                    last_index = current.lastIndexOf "/"
+                    last_index = current.last_index_of "/"
                     if last_index == Nothing then (S3_Error.Error "Cannot move above root folder.") else current.take last_index
                 "." -> current
-                x -> new_current + "/" + x
+                x -> extend current x
```

Comment:
This is getting complicated.

I think we should encapsulate the path handling logic to not be part of `S3_File` but some `S3_Path_Helper`. I did not want to complicate this PR though, so ideally I'd do this as a separate ticket if we decide it's worth it.

Reply:
Followup: #9021